lib/post.js

```diff
@@ -4,43 +4,38 @@
 "use strict";
 
 var extend = require("lodash").extend;
-var execute = require("./task");
-var http = require("http");
 var xpi = require("./xpi");
-var cp = require('child_process');
-var fs = require('fs');
-var url = require('url');
-var net = require('net');
+var fs = require("fs");
+var url = require("url");
+var net = require("net");
 
 function postXPI (manifest, options) {
   var postURL = options.postUrl;
 
-  return xpi(manifest, options).then(function (xpiPath) {
-    console.log("Successfully created xpi at " + xpiPath);
-
-    fs.readFile(xpiPath, null, function(err, buffer) {
-      var p = url.parse(postURL);
-      var client = net.connect({
-        host: p.hostname,
-        port: p.port
-      }, function() {
-        var identifier = new Buffer('\rPOST / HTTP/1.1\n\rUser-Agent: NodeJS Compiler\n\r\n');
-        client.write(identifier);
-        client.end(buffer);
-      });
-      client.on('data', function(data) {
-        // Extension Auto-Installer does not close the connection, so we do it after 500ms
-        setTimeout(function() {
-          client.destroy();
-        }, 500);
-      });
-      client.on('error', function (e) {
-        // if net errors need to be propagated.
-      });
-      client.on('close', function (had_error) {
-        console.log(had_error ? "Posting XPI to %s failed" : "Posted XPI to %s", postURL);
-        fs.unlink(xpiPath);
-        console.log("Removed XPI from " + xpiPath);
+  return xpi(manifest, extend(options, { skipUpdateRDF: true })).then(function (xpiPath) {
+		return when.promise(function(done) {
+      fs.readFile(xpiPath, function(err, buffer) {
+        var p = url.parse(postURL);
+        var client = net.connect({ host: p.hostname, port: p.port }, function() {
+          var identifier = new Buffer('\rPOST / HTTP/1.1\n\rUser-Agent: NodeJS Compiler\n\r\n');
+          client.write(identifier);
+          client.end(buffer);
+        });
+        client.on('data', function (data) {
+          // Extension Auto-Installer does not close the connection, so we do it after 500ms
+          setTimeout(function() {
+            client.destroy();
+          }, 500);
+        });
+        client.on('error', function (e) {
+          // if net errors need to be propagated.
```

Comment:
while you're in here, you might as well change the signature to `when.promise(function(resolve, reject) {...` so you can call `reject(e)` right here. That will show a stack trace in the console and will allow the caller to handle the promise rejection.